### PR TITLE
DOCSP-38417 - Aggregation

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-38417
Staging - https://preview-mongodbmongokart.gatsbyjs.io/pymongo/docsp-38417-agg-funds/aggregation/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/aggregation.txt [hunk 1]
```diff
@@ -1,17 +1,265 @@
...
+   memory limit of 100 megabytes and ignores the ``allowDiskUse`` parameter.
+
+Aggregation Example
+-------------------
+
+.. note::
+  
+  This example uses the ``sample_restaurants.restaurants`` collection
+  from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+  free MongoDB Atlas cluster and load the sample datasets, see the :ref:`<pymongo-get-started>`.
```
Comment (jordan-smith721) [inline source/aggregation.txt:91]:
```suggestion
  free MongoDB Atlas cluster and load the sample datasets, see :ref:`<pymongo-get-started>`.
```

File: source/aggregation.txt [hunk 2]
```diff
@@ -1,17 +1,265 @@
...
+
+The following code example produces a count of the number of bakeries in each borough
+of New York. To do so, it uses an aggregation pipeline with the following stages:
+
+- A :manual:`$match </reference/operator/aggregation/match/>` stage to filter for documents
+  whose ``cuisine`` field contains the value ``"Bakery"``.
+
+- A :manual:`$group </reference/operator/aggregation/group/>` stage to group the matching
+  documents by the ``borough`` field, accumulating a count of documents for each distinct
+  value of ``borough``.
```
Comment (jordan-smith721) [inline source/aggregation.txt:104]:
```suggestion
  value.
```

File: source/aggregation.txt [hunk 3]
```diff
@@ -1,17 +1,265 @@
...
+   {'_id': 'Missing', 'count': 2}
+   {'_id': 'Manhattan', 'count': 221}
+   {'_id': 'Queens', 'count': 204}
+   {'_id': 'Staten Island', 'count': 20}
+
+Explain an Aggregation
+~~~~~~~~~~~~~~~~~~~~~~
+
+To view information about how MongoDB executes your operation, you can
+ask MongoDB to **explain** it. When MongoDB explains an operation, it returns
```
Comment (jordan-smith721) [inline source/aggregation.txt:137]: S: This is nitpicky but I think "prompt" is more accurate than "ask" here

```suggestion
prompt MongoDB to **explain** it. When MongoDB explains an operation, it returns
```

File: source/aggregation.txt [hunk 4]
```diff
@@ -1,17 +1,265 @@
...
+**execution plans** and performance statistics. An execution
+plan is a potential way MongoDB can complete an operation.
+When you instruct MongoDB to explain an operation, it returns both the winning execution
+plan, which is the plan MongoDB executed, and any rejected execution plans.
```
Comment (jordan-smith721) [inline source/aggregation.txt:141]: S: Simplify

```suggestion
When you instruct MongoDB to explain an operation, it returns both the plan MongoDB executed, and any rejected execution plans.
```

File: source/aggregation.txt [hunk 5]
```diff
@@ -1,17 +1,265 @@
...
+To explain an aggregation operation, you can use either the
+`PyMongoExplain <https://pypi.org/project/pymongoexplain/>`__ library or a database
+command. Select the corresponding tab below to see an example of each method.
+
+.. tabs::
+
+   .. tab:: PyMongoExplain
+      :tabid: pymongoexplain
+
+      You must use pip to install the ``pymongoexplain`` library, as shown in the
```
Comment (jordan-smith721) [inline source/aggregation.txt:152]:
```suggestion
      Use pip to install the ``pymongoexplain`` library, as shown in the
```

**Review by jordan-smith721 — APPROVED** [review]
LGTM w/ a few suggestions

File: source/aggregation.txt [hunk 6]
```diff
@@ -1,17 +1,265 @@
...
+Transform Your Data with Aggregation
+====================================
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :keywords: code example, transform, computed, pipeline
+   :description: Learn how to use (+driver-short+} to perform aggregation operations.
```
[the remark this answers is not in the document]
Comment (mongoKart) [reply source/aggregation.txt:13]: good catch

File: source/aggregation.txt [hunk 7]
```diff
@@ -1,17 +1,265 @@
...
+assembly stations, and **operator expressions** are the
+specialized tools.
+
+Aggregation vs. Find Operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use find operations to perform the following actions:
+
+- Select *what* documents to return
+- Select *what* fields to return
```
[the remark this answers is not in the document]
Comment (mongoKart) [reply source/aggregation.txt:55]: You're right, that's a weird choice

File: source/aggregation.txt (hunk 3 above)
Comment (mongoKart) [reply source/aggregation.txt:137]: used "instruct"

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**